plugins/hls-semantic-tokens-plugin/src/Ide/Plugin/SemanticTokens/Utils.hs

```diff
@@ -0,0 +1,94 @@
+{-# LANGUAGE StandaloneDeriving #-}
+{-# LANGUAGE TypeFamilies       #-}
+{-# OPTIONS_GHC -Wno-orphans #-}
+
+
+module Ide.Plugin.SemanticTokens.Utils where
+
+import           Data.ByteString                 (ByteString)
+import           Data.ByteString.Char8           (unpack)
+import qualified Data.Map                        as Map
+import           Development.IDE.GHC.Compat
+import           Ide.Plugin.SemanticTokens.Types
+import           Prelude                         hiding (span)
+
+deriving instance Show DeclType
+deriving instance Show BindType
+deriving instance Show RecFieldContext
+
+instance Show ContextInfo where
```

Comment:
can we upstream this?

Reply:
I am keeping this here for convenient of development, the upstream have [Outputable](https://hackage.haskell.org/package/ghc-9.8.1/docs/src/GHC.Utils.Outputable.html#Outputable) for this.

Reply:
Ah, I didn't realise this was a GHC type. And I guess the outputable instance isn't useful? We use that for other types here...

Reply:
outputable is too user friendly(Being verbose), I just want a more compact output that fits the source better here for debugging